packages/graphql/src/schema/make-augmented-schema.ts

```diff
@@ -501,20 +530,22 @@ function makeAugmentedSchema({
         ...(hasGlobalNodes ? { Node: { __resolveType: (root) => root.__resolveType } } : {}),
     };
 
-    unionTypes.forEach((union) => {
-        // It is possible to make union types "writeonly". In this case adding a resolver for them breaks schema generation.
-        const unionTypeInSchema = parsedDoc.definitions.find((def) => {
-            if (def.kind === Kind.UNION_TYPE_DEFINITION && def.name.value === union.name.value) return true;
+    schemaModel.compositeEntities.forEach((compositeEntity) => {
```

Comment:
Here it's iterating over the whole compositeEntities and then again the whole definitions, this could have an impact on the performance for type definitions with a large number of composite entities!

Reply:
yeah it's not ideal, however this is pretty much how it was before (before it was looping through all the unions and all the interfaceRelationships, now it's looping through all composite entities so the only additional ones are interfaces that are not used as targets of relationships, which are usually a very little minority). 
However this is a bigger issue to do with the entire schema generation flow that shouldn't be in the scope of this PR.
I left a TODO comment to bring awareness to the issue.

Reply:
Sounds good, didn't expected a resolution just wanted to raise awareness of the bottleneck!